includes/checkout/functions.php

```diff
@@ -281,8 +281,16 @@ function edd_is_email_banned( $email = '' ) {
 		return false;
 	}
 
-	$ret = in_array( trim( $email ), edd_get_banned_emails() );
+	$banned_emails = edd_get_banned_emails();
 
+	foreach( $banned_emails as $banned_email ) {
+		if( is_email( $banned_email ) ) {
+			$ret = ( $banned_email == trim( $email ) ? true : false );
```

Comment:
@pippinsplugins I think it simplifies it, but for what it's worth, the readability of it as it stands is better I think.